### ISSUE TITLE
False Positive `DTZ005`: When using `datetime.datetime.now().astimezone()`

### ISSUE THREAD

**Issue by LawrenceJGD** [astral-sh/ruff#5516]
The DTZ005 rule of flake8-datetimez throws an error when using the `datetime.datetime.now()` class method without the `tz` argument in `now()` or when the argument is `None`, it does this to avoid problems with `datetime.datetime` objects without a timezone, but there is a way to get the local time and date using `datetime. datetime.now().astimezone()`, when astimezone() detects that the object does not have a timezone then it assigns the local one, but still flake8-datetimez detects this as an error, so it would be better to fix it, since in the end the object ends up having a timezone.

This is referenced by pjknkda/flake8-datetimez#9 and #1249.

The command I use is `ruff check test.py` and the example source code is this:

```python3
import datetime

datetime.datetime.now().astimezone()
```

I'm using ruff 0.0.269 and my `pyproject.toml`:

```toml
[tool.ruff]
select = [
    "W", "C90", "I", "N", "D", "UP", "YTT", "ANN", "S", "BLE", "FBT", "B", "A",
    "COM", "C4", "DTZ", "ICN", "PIE", "RSE", "RET", "SLF", "SIM", "TCH", "RUF",
    "INT", "ARG", "TD", "EM", "PLC", "PLE", "PLR", "PLW", "TRY", "FLY", "E",
    "F"
]
line-length = 88
target-version = "py38"
ignore = ["COM812"]

[tool.ruff.pydocstyle]
convention = "google"

[tool.black]
line-length = 88
target-version = ["py38", "py39", "py310", "py311"]
```

**Comment by charliermarsh** [comment]
Seems reasonable to me!

**Comment by dhruvmanila** [comment]
Current implementation doesn't check if `astimezone` is called with or without any arguments i.e., it allows either of them.

Could you provide an example of why `datetime.datetime.now().astimezone(mytz)` usage is considered bad?

**Comment by LawrenceJGD** [comment]
I think that if you want to create a `datetime.datetime` object with the current time using another time zone than the local one, it would be better to use `datetime.datetime.now(mytz)`, but is seems that this and `datetime.datetime.now().astimezone(mytz)` are both valid ways for achieving the same thing.

BTW, thanks for being so fast solving this!

**Comment by bluetech** [comment]
You're right, `datetime.datetime.now().astimezone(mytz)` ends up being OK as well, because `now()` is naive local, and `astimezone` assumes naive is local.

Here's my analysis on the yes-arguments case:

- :x: DTZ001 call-datetime-without-tzinfo, I've seen people do `datetime.datetime(2023, 7, 6, 2, 5).astimezone(mytz)` when they intended to do `datetime.datetime(2023, 7, 6, 2, 5, tzinfo=mytz)` (or `replace(tzinfo=mytz)`). IMO there is no good reason to use `astimezone(mytz)` here.

- :heavy_check_mark: DTZ002 call-datetime-today, local time so OK.

-  :x: DTZ003 call-datetime-utcnow, using `astimezone()` is going to be wrong always I believe (unless you live in Greenwich). `utcnow()` itself is deprecated now for being a footgun.

- :x: DTZ004 call-datetime-utcfromtimestamp, same as DTZ003.

- :heavy_check_mark: DTZ005 call-datetime-now-without-tzinfo, local time so OK.

- :question: DTZ006 call-datetime-fromtimestamp, never used this function, not sure

- ⍻ DTZ007 call-datetime-strptime-without-zone, depends on the input string. If it's in local time, usage is OK. If not, it's a bug. I suppose in such cases ruff should allow. From the rule description I think it's already allowed so OK.

**Comment by dhruvmanila** [comment]
You're correct in your analysis but I think the diagnostics are correct, it's the _usage_ which might be incorrect from the user side. I don't think we can assume anything about the usage. Correct me if I'm wrong but I think the goal for this rule is to detect if the initialized `datetime.datetime` object is timezone aware which will be the case if `astimezone` was called or if it was provided during initialization.
